crate/pages/blog/managing-crate-with-mesos.html

```diff
@@ -0,0 +1,148 @@
+title: Running and managing Crate databases with Mesos
+author: Chris Ward
+description: This post will give you a step-by-step guide to running and managing Crate databases with our Mesos integration framework.
+created: 2015/05/13 12:10:00
+post_name: managing-crate-with-mesos
+status: publish
+post_type: post
+tags: mesos,mesosphere
+category: news, developer-news
+
+Apache Mesos is a fantastic tool for abstracting CPU, memory, storage, and other compute resources away from machines (physical or virtual). This lets you program against your datacenter like it's a single pool of resources.
+
+Crate's integration with Mesos enables simple management across any number of Crate instances you may require without needing explicit knowledge on the quantity and their specifications.
+
+The integration framework offers a variety of configuration and management options such as dynamically increasing cluster sizes, upgrades, data locations and using Mesos-DNS for service discovery.
+
+## 1. Getting Started
+You will need to have a functioning Mesos installation for the rest of this tutorial and it will be assumed that you do. If you don't, then [find full instructions here](https://docs.mesosphere.com/getting-started).
+
+### Users and Roles
+The Crate Framework runs as the _crate_ user and _*_ role by default. This is configurable using `--framework-user` and `--framework-role` if required.
+
+This means that this user needs to be present on all instances, both master and slaves. The user does not need any specific permissions. Add a user with:
+
+`useradd crate -s /bin/bash`
+
+### Persistent Data Paths
+If you want to utilize [Persistent Data Paths](https://github.com/crate/crate-mesos-framework#persistent-data-paths) (recommended), you need to make sure that the user _crate_ (or the user you decided on above) has write permissions to these locations. For example:
+
+`chown crate:crate /path/to/persistent/disk`
+
+### Ports
+By default Crate uses the ports 4200 and 4300. To receive resource offers from Mesos you need to add the resource reservation for a port range that includes these ports. This happens when starting the framework and can be done by writing to the resources file:
+
+```
+echo 'ports(*)[31000-31099, 31101-32000, 4200-4399]' > /etc/mesos-slave/resources
+```
+
+Or by starting the slave with the option:
+
+```
+--resources=ports(*)[31000-31099, 31101-32000, 4200-4399]
+```
+
+Restart the slave and clean the old slave state if necessary (`rm -f /tmp/mesos/meta/slaves/latest`).
+
+## 2. Installation
+### Manual Installation (Not Recommended)
```

Comment:
There are 2 ways of installing the Crate Mesos Framework: manually, or via Marathon. The latter is recommended.